Backend/filesystem/service.go

```diff
@@ -97,20 +97,21 @@ func renameEntity(pool database.Pool, entityID int, newName string) error {
 	return err
 }
 
-// // getFilesystemInfo returns information regarding a specific file system entity
-// func getEntityChildren(pool database.Pool, docID int) (EntityInfo, error) {
-// 	entity := EntityInfo{}
-// 	children := pgtype.Hstore{}
-
-// 	err := pool.GetConn().QueryRow(context.Background(), "SELECT EntityID FROM filesystem WHERE Parent = $1",
-// 		docID).Scan(&entity.EntityID, &entity.EntityName, &entity.IsDocument, &children)
-// 	if err != nil {
-// 		return EntityInfo{}, errors.New("failed to read from database")
-// 	}
-
-// 	for k := range children.Map {
-// 		entity.Children = append(entity.Children, k)
-// 	}
-
-// 	return entity, nil
-// }
+// getFilesystemInfo returns information regarding a specific file system entity
+func getEntityChildren(pool database.Pool, docID int) ([]string, error) {
+	children := pgtype.Hstore{}
+
+	err := pool.GetConn().QueryRow(context.Background(), "SELECT children FROM filesystem WHERE entityid = $1",
+		docID).Scan(&children)
+	if err != nil {
+		return nil, errors.New("failed to read from database")
+	}
+
+	list := []string{}
+
+	for k := range children.Map {
+		list = append(list, k)
```

Comment:
This isn't what `getEntityChildren` should be doing, the functionality is close though. When the frontend calls the getEntityChildren endpoint ideally we should also return some basic information regarding each child (not just the ID), that way they don't have to repeatedly flood the backend with more requests when trying to capture information about each child. Ideally the response should look something like:
```json
{
   "children": [
      {
          "entityID": 1234,
          "name": "cool document",
          "isDocument": true
      },
      {
          "entityID": 1234,
          "name": "cool directory",
          "isDocument": false
      },
   ]
}
```

Reply:
so something roughly like:
```go
children := []EntityInfo{}

for k := range children.Map {
   info, _ := getFilesystemInfo(pool, k)
   children = append(children, info)
}
return children
```

Reply:
oooh my bad, will update to match the api docs

Reply:
I've changed it to return entityinfo now, although that also means including an array of the children of the children since it's using the output from getFilesystemInfo